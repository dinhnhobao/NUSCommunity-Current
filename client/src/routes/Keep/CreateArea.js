import React from "react";
import AddIcon from "@material-ui/icons/Add";
import Fab from "@material-ui/core/Fab";
import Zoom from "@material-ui/core/Zoom";

export default class CreateArea extends React.Component {
  state = {
    isExpanded: false,
    note: {
      title: "",
      content: "",
    },
  };

  handleChange = (event) => {
    const { name, value } = event.target;
    const { note } = this.state;
    this.setState({
      note: {
        ...note,
        [name]: value,
      },
    });
  };

  submitNote = (event) => {
    const { note } = this.state;
    const { onAdd } = this.props;
    onAdd(note);
    this.setState({
      note: {
        title: "",
        content: "",
      },
    });
    event.preventDefault();
  };

  expand = () => {
    this.setState({
      isExpanded: true,
    });
  };

  render() {
    const { isExpanded, note } = this.state;
    return (
      <div>
        <form className="create-note">
          {isExpanded && (
            <input
              name="title"
              onChange={this.handleChange}
              value={note.title}
              placeholder="Title"
            />
          )}
          <textarea
            name="content"
            onClick={this.expand}
            onChange={this.handleChange}
            value={note.content}
            placeholder="Take a note..."
            rows={isExpanded ? 3 : 1}
          />
          <Zoom in={isExpanded}>
            <Fab onClick={this.submitNote}>
              <AddIcon />
            </Fab>
          </Zoom>
        </form>
      </div>
    );
  }
}
