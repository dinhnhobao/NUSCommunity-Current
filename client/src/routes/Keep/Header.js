import React from "react";
import EmojiObjectsIcon from "@material-ui/icons/EmojiObjects";

export default class Header extends React.Component {
  render() {
    return (
      <header>
        <h1>
          <EmojiObjectsIcon />
          Keep
        </h1>
      </header>
    );
  }
}
