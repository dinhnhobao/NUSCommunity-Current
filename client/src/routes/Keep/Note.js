import React from "react";
import DeleteIcon from "@material-ui/icons/Delete";

export default class Note extends React.Component {
  handleClick = () => {
    const { onDelete, id } = this.props;
    onDelete(id);
  };

  render() {
    const { title, content } = this.props;
    return (
      <div className="note">
        <h1>{title}</h1>
        <p>{content}</p>
        <button onClick={this.handleClick}>
          <DeleteIcon />
        </button>
      </div>
    );
  }
}
