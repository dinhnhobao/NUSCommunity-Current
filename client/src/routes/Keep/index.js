import React from "react";
// import Header from "./Header";
import Note from "./Note";
import CreateArea from "./CreateArea";
import "./index.css";

export default class Keep extends React.Component {
  state = {
    notes: [],
  };

  addNote = (newNote) => {
    const { notes } = this.state;
    this.setState({
      notes: [...notes, newNote],
    });
  };

  deleteNote = (id) => {
    const { notes } = this.state;
    this.setState({
      notes: notes.filter((noteItem, index) => {
        return index !== id;
      }),
    });
  };

  render() {
    const { notes } = this.state;
    return (
      <div>
        <CreateArea onAdd={this.addNote} />
        {notes.map((noteItem, index) => {
          return (
            <Note
              key={index}
              id={index}
              title={noteItem.title}
              content={noteItem.content}
              onDelete={this.deleteNote}
            />
          );
        })}
      </div>
    );
  }
}
