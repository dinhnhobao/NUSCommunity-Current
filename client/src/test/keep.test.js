import React from 'react';
import {shallow} from 'enzyme';
import Keep from '../routes/Keep/index';

const setUp = (props={}) => {
    const component = shallow(<Keep {...props} />);
    return component;
}

const findByTestAttr = (component, attr) => {
    const wrapper = component.find(`[data-test='${attr}']`);
    return wrapper;
}

describe('Keep Component', () => {
    let component;
    beforeEach(() => {
        component = setUp();
    })

    it('Should render correctly', () => {
        const wrapper =  findByTestAttr(component, 'keepComponent');
        expect(wrapper).toMatchSnapshot();
    });

    it('Should render without errors', () => {
        const wrapper =  findByTestAttr(component, 'keepComponent');
        expect(wrapper.length).toEqual(1);       
    });

    describe('Create Add Area', () => {
        it('Should render without errors', () => {
            const wrapper =  findByTestAttr(component, 'addArea');
            expect(wrapper.length).toEqual(1);
        });
    })
})
