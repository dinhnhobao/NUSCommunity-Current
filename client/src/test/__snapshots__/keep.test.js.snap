// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keep Component Should render correctly 1`] = `ShallowWrapper {}`;
